src/Compilers/Core/Portable/NativePdbWriter/ComMemoryStream.cs

```diff
@@ -52,6 +63,32 @@ public void CopyTo(Stream stream)
             }
         }
 
+        public IEnumerable<ArraySegment<byte>> GetChunks()
```

Comment:
>ArraySegment<byte> [](start = 27, length = 18)

It feels strange that we are exposing underlying arrays to outside consumer, which is free to modify content in the arrays.

Reply:
There is no harm in that. The stream is read/write, you can change any byte by calling Write() method. GetChunks just provides a way to enumerate all content without making copies.

---
In reply to: [137341658](https://github.com/dotnet/roslyn/pull/21712#discussion_r137341658) [](ancestors = 137341658)

Reply:
It looks like modifications to the last returned array could be ignored.